nemo/collections/nlp/modules/common/transformer/transformer.py

```diff
@@ -0,0 +1,151 @@
+    num_token_types: int = 2
+    embedding_dropout: float = 0.0
+    learn_positional_encodings: bool = False
+
+    # transformer
+    num_attention_heads: int = 1
+    ffn_dropout: float = 0.0
+    attn_score_dropout: float = 0.0
+    attn_layer_dropout: float = 0.0
+    hidden_act: str = 'relu'
+    pre_ln: bool = False
+
+
+@dataclass
+class TransformerEncoderConfig(TransformerConfig):
+    mask_future: bool = False
+
+
+class TransformerEncoderNM(EncoderModule):
+    def __init__(
+        self,
+        vocab_size: int,
+        hidden_size: int,
+        num_layers: int,
+        inner_size: int,
+        max_sequence_length: int = 512,
+        num_token_types: int = 2,
+        embedding_dropout: float = 0.0,
+        learn_positional_encodings: bool = False,
+        num_attention_heads: int = 1,
+        ffn_dropout: float = 0.0,
+        attn_score_dropout: float = 0.0,
+        attn_layer_dropout: float = 0.0,
+        hidden_act: str = 'relu',
+        mask_future: bool = False,
+        pre_ln: bool = False,
+    ):
+        super().__init__()
+
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+
+        self.embedding = TransformerEmbedding(
+            vocab_size=vocab_size,
+            hidden_size=hidden_size,
+            max_sequence_length=max_sequence_length,
+            num_token_types=num_token_types,
+            embedding_dropout=embedding_dropout,
+            learn_positional_encodings=learn_positional_encodings,
+        )
+
+        self.encoder = TransformerEncoder(
+            hidden_size=hidden_size,
+            num_layers=num_layers,
+            inner_size=inner_size,
+            num_attention_heads=num_attention_heads,
+            ffn_dropout=ffn_dropout,
+            attn_score_dropout=attn_score_dropout,
+            attn_layer_dropout=attn_layer_dropout,
+            hidden_act=hidden_act,
+            mask_future=mask_future,
+            pre_ln=pre_ln,
+        )
+
+    def forward(self, input_ids, encoder_mask):
+        embeddings = self.embedding(input_ids)
+        encoder_hidden_states = self.encoder(embeddings, encoder_mask)
+        return encoder_hidden_states
+
+
+class TransformerDecoderNM(DecoderModule):
+    def __init__(
+        self,
+        vocab_size: int,
+        hidden_size: int,
+        num_layers: int,
+        inner_size: int,
+        max_sequence_length: int = 512,
+        num_token_types: int = 2,
+        embedding_dropout: float = 0.0,
+        learn_positional_encodings: bool = False,
+        num_attention_heads: int = 1,
+        ffn_dropout: float = 0.0,
+        attn_score_dropout: float = 0.0,
+        attn_layer_dropout: float = 0.0,
+        hidden_act: str = 'relu',
+        pre_ln: bool = False,
+    ):
+        super().__init__()
+
+        self.hidden_size = hidden_size
+        self.max_sequence_length = max_sequence_length
+
+        self.embedding = TransformerEmbedding(
+            vocab_size=vocab_size,
```

Comment:
please pass here:
```
vocab_size=self.vocab_size
hidden_size=self.hidden_size
```
would make it a little less error-prone when others add changes to these files.

Reply:
Done.